### PR TITLE
BGDIINF_SB-1482: Added Item POST, PUT, PATCH, DELETE Endpoints

### PR DESCRIPTION
**Pull request by ltshb**
Implemented transactional API for Items.

Added new `views_mixins` to handle correctly the DELETE with a JSON answer and to handle the `Location` header.

**NOTES:**
- ETag will be done in a separate PR
- SPEC update will be done in a separate PR
- This PR requires synchronization with #131 and #132

### CODE REVIEW

File: app/stac_api/models.py [hunk 1]
```diff
@@ -363,7 +408,10 @@ class Item(models.Model):
     # properties_license = models.TextField(blank=True)
     # properties_platform = models.TextField(blank=True)
     # properties_providers = models.ManyToManyField(Provider)
-    properties_title = models.CharField(blank=True, max_length=255)
+    # Although it is discouraged by Django to set blank=True and null=True on a CharField, it is
```
Comment (rebert) [inline app/stac_api/models.py:411]: thanks for investigating here!

**Review by boecklic — APPROVED** [review]
Nice! Looks good as far as I see

**Review by rebert — CHANGES_REQUESTED** [review]
Very nice.

I was doing some functional tests:
make serve:
When adding a new item via form and I only fill out an item id, I get the followin error:
`

Request Method: | POST
-- | --
http://localhost:8000/api/stac/v0.9/collections/ch.swisstopo.swisstlmregio/items
3.1.3
AttributeError
This QueryDict instance is immutable
/home/ltrea/service_stac/.venv/lib/python3.7/site-packages/django/http/request.py, line 488, in _assert_mutable
/home/ltrea/service_stac/.venv/bin/python3
3.7.5
['/home/ltrea/service_stac/app',  '/usr/lib/python37.zip',  '/usr/lib/python3.7',  '/usr/lib/python3.7/lib-dynload',  '/home/ltrea/service_stac/.venv/lib/python3.7/site-packages']
Wed, 09 Dec 2020 07:39:56 +0000

`

**Comment by ltshb** [comment]
> This QueryDict instance is immutable

Good catch, I had this issue once with the unittest but then it went away, strange that unit test are now working but not the real POST. I'll investigate. Thanks for testing.

**Review by rebert — APPROVED** [review]

**Pull request merged**